Bottom: 2c591c65c7bd9f57ae78923fe5de8cfe049f93a1
Top:    7f3c11593a41bb2063e210da3da48636c0b982dc
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-03-21 12:03:37 +0000

polygon.c: For some reason, the range computation in frac_circle appears to be wrong


---

diff --git a/src/polygon.c b/src/polygon.c
index ce68a11..6a63c0d 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -405,8 +405,8 @@ frac_circle (PLINE * c, Coord X, Coord Y, Vector v, int fraction)
   e1 = (v[0] - X) * POLY_CIRC_RADIUS_ADJ;
   e2 = (v[1] - Y) * POLY_CIRC_RADIUS_ADJ;
 
-  /* NB: the caller adds the last vertex, hence the -1 */
-  range = POLY_CIRC_SEGS / fraction - 1;
+  /* XXX */ /* NB: the caller adds the last vertex, hence the -1 */
+  range = POLY_CIRC_SEGS / fraction;
   for (i = 0; i < range; i++)
     {
       /* rotate the vector */
